### PR TITLE
Error for unsupported flag combo

### PR DESCRIPTION
**Pull request by sukhil-suresh**
- Error when running `create-bui0lder` and `package-buildpack` with both --no-pull and --publish
- Error with `The --publish and --no-pull flags cannot be used together. There is nothing to pull if you are publishing.`

Signed-off-by: Sukhil Suresh <ssukhil@vmware.com>

### CODE REVIEW

File: internal/commands/create_builder_test.go [hunk 1]
```diff
@@ -53,6 +53,20 @@ func testCreateBuilderCommand(t *testing.T, when spec.G, it spec.S) {
...
+			it("errors with a descriptive message", func() {
+				command.SetArgs([]string{
+					"some/builder",
+					"--builder-config", "some-config-path",
+					"--publish",
+					"--no-pull",
+				})
+				err := command.Execute()
+				h.AssertNotNil(t, err)
+				h.AssertError(t, err, "The --publish and --no-pull flags cannot be used together. There is nothing to pull if you are publishing.")
```
Comment (jromero) [inline internal/commands/create_builder_test.go:66]:
```suggestion
				h.AssertError(t, err, "The --publish and --no-pull flags cannot be used together. The --publish flag requires the use of remote images.")
```

File: internal/commands/package_buildpack.go [hunk 2]
```diff
@@ -19,6 +19,13 @@ type PackageBuildpackFlags struct {
 	NoPull          bool
 }
 
+func (p PackageBuildpackFlags) validate() error {
+	if p.Publish && p.NoPull {
+		return errors.Errorf("The --publish and --no-pull flags cannot be used together. There is nothing to pull if you are publishing.")
```
Comment (jromero) [inline internal/commands/package_buildpack.go:24]:
```suggestion
		return errors.Errorf("The --publish and --no-pull flags cannot be used together. The --publish flag requires the use of remote images.")
```

File: internal/commands/package_buildpack_test.go [hunk 3]
```diff
@@ -27,6 +27,26 @@ func TestPackageBuildpackCommand(t *testing.T) {
...
+				command.SetArgs([]string{
+					"some-image-name",
+					"--package-config", "/path/to/some/file",
+					"--publish",
+					"--no-pull",
+				})
+
+				err := command.Execute()
+				h.AssertNotNil(t, err)
+				h.AssertError(t, err, "The --publish and --no-pull flags cannot be used together. There is nothing to pull if you are publishing.")
```
Comment (jromero) [inline internal/commands/package_buildpack_test.go:46]:
```suggestion
				h.AssertError(t, err, "The --publish and --no-pull flags cannot be used together. The --publish flag requires the use of remote images.")
```

File: internal/commands/create_builder.go [hunk 4]
```diff
@@ -18,13 +18,24 @@ type CreateBuilderFlags struct {
 	NoPull          bool
 }
 
+func (c CreateBuilderFlags) validate() error {
+	if c.Publish && c.NoPull {
+		return errors.Errorf("The --publish and --no-pull flags cannot be used together. There is nothing to pull if you are publishing.")
```
Comment (jromero) [inline internal/commands/create_builder.go:23]:
```suggestion
		return errors.Errorf("The --publish and --no-pull flags cannot be used together. The --publish flag requires the use of remote images.")
```

**Comment by sukhil-suresh** [comment]
@jromero Made another commit to update error message. Thanks for the quick feedback!

**Pull request merged**